micrometer-core/src/test/java/io/micrometer/core/aop/TimedAspectTest.java

```diff
@@ -370,6 +376,58 @@ void timeClassFailure() {
         });
     }
 
+    @Test
+    void metricTagsWithText() {
+        MeterRegistry registry = new SimpleMeterRegistry();
+        TimedAspect timedAspect = new TimedAspect(registry);
+        timedAspect.setMetricsTagAnnotationHandler(
+                new MetricsTagAnnotationHandler(aClass -> tagValueResolver, aClass -> tagValueExpressionResolver));
+
+        AspectJProxyFactory pf = new AspectJProxyFactory(new MetricTagClass());
+        pf.addAspect(timedAspect);
+
+        MetricTagClass service = pf.getProxy();
+
+        service.getAnnotationForArgumentToString(15L);
+
+        assertThat(registry.get("method.timed").tag("test", "15").timer().count()).isEqualTo(1);
+    }
+
+    @Test
+    void metricTagsWithResolver() {
+        MeterRegistry registry = new SimpleMeterRegistry();
+        TimedAspect timedAspect = new TimedAspect(registry);
+        timedAspect.setMetricsTagAnnotationHandler(
+                new MetricsTagAnnotationHandler(aClass -> tagValueResolver, aClass -> tagValueExpressionResolver));
+
+        AspectJProxyFactory pf = new AspectJProxyFactory(new MetricTagClass());
+        pf.addAspect(timedAspect);
+
+        MetricTagClass service = pf.getProxy();
+
+        service.getAnnotationForTagValueResolver("foo");
```

Comment:
It'd be good to test that the tag value can change based on the parameter value. Even just concatenating it onto the current constant string would be good, I think.

Reply:
Most likely I didn't understand what you're saying but is there any difference between `("foo")` and `(FO + "o")`?

Reply:
I meant that we aren't testing that the value passed to the annotated method parameter (in this case, `"foo"`) is passed to the value resolver because the value resolver used for the test returns a constant string. So if there were a bug that the value passed to the value resolver were always `null` or something, this test would still pass.